src/probnum/randprocs/kernels/_kernel.py

```diff
@@ -495,8 +423,8 @@ class IsotropicMixin(abc.ABC):  # pylint: disable=too-few-public-methods
     def _squared_euclidean_distances(
         self, x0: np.ndarray, x1: Optional[np.ndarray]
     ) -> np.ndarray:
-        """Implementation of the squared Euclidean distance, which supports kernel
-        broadcasting semantics."""
+        """Implementation of the squared Euclidean distance, which supports scalar
```

Comment:
Why scalar inputs? Doesnt this compute pairwise squared euclidean distances between _vectors_? Same for the related functions below.

Reply:
Conceptually yes, but in order to unify implementation, we also need to be able to handle the base-cases `input_shape=(1,)` and `input_shape=()`. The latter is technically a scalar.